cmd/register/register.go

```diff
@@ -0,0 +1,24 @@
+package register
+
+import (
+	cmdcore "github.com/lyft/flytectl/cmd/core"
+	"github.com/spf13/cobra"
+)
+
+// RegisterCommand will return register command
+func RegisterCommand() *cobra.Command {
+	registerCmd := &cobra.Command{
+		Use:   "register",
+		Short: "Registers tasks/workflows/launchplans from list of generated serialized files.",
+		Long: "Takes input files as serialized versions of the tasks/workflows/launchplans and registers them with flyteadmin.\n" +
+			  "Currently these input files are protobuf files generated as output from flytekit serialize.\n" +
+			  "Project & Domain are mandatory fields to be passed for registration and an optional version which defaults to v1\n" +
+			  "If the entities are already registered with flyte for the same version then registration would fail.\n" ,
+	}
+	registerResourcesFuncs := map[string]cmdcore.CommandEntry{
+		"files":    {CmdFunc: registerFromFilesFunc, Aliases: []string{"file"}},
+	}
+	cmdcore.AddCommands(registerCmd, registerResourcesFuncs)
+	registerCmd.PersistentFlags().StringVarP(&(GetConfig().version), "version", "v", "v1", "Specifies the Version to use for registering the serialized files.")
```

Comment:
no need if you `go generate`

Reply:
Done

Reply:
The global flag of --register.version along with other global flags are not getting updated in the Config which is needed by the command actions. I am going back to using persistent flags . Will create an issue to fix this across the board.
Having a persistent flag has an advantage that this flag is only visible under the command context